NEWS.md

```diff
@@ -1,5 +1,12 @@
 # vctrs (development version)
 
+* Fixed an issue where certain ALTREP row names were being materialized when
+  passed to `new_data_frame()`. We've fixed this by removing a safeguard in
+  `new_data_frame()` that performed a compatibility check when both `n` and
+  `row.names` were provided. Because this is a low level function designed for
```

Comment:
The docs for `new_data_frame()` actually say:

```
It is meant to be performant, and does not check the inputs for correctness in any way
```

so it really doesn't bother me much to remove this safeguard, especially if we document that a `row.names` attribute will override both a user supplied `n` and the implied size of `x`, which I have done

---

There are also plenty of other ways to make incompatible data frames already

```
new_data_frame(list(x = 1:2), row.names = 1:3)
new_data_frame(list(x = 1:2), n = 3)
```